server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java

```diff
@@ -107,7 +108,8 @@ private ValuesWsResponse doHandle(Request request) {
       Optional<ComponentDto> component = getComponent(dbSession, valuesRequest);
       checkAdminPermission(component);
       Set<String> keys = new HashSet<>(valuesRequest.getKeys());
-      return new ValuesResponseBuilder(loadSettings(dbSession, component, keys), component).build();
+      Map<String, String> deprecatedKeysByValidKeys = getDeprecatedKeysByValidKeys(keys);
```

Comment:
do you think it's better to manage the deprecate key mechanism in the SettingsLoader?